cmd/restic/global.go

```diff
@@ -111,7 +111,7 @@ func init() {
 	f.CountVarP(&globalOptions.Verbose, "verbose", "v", "be verbose (specify multiple times or a level using --verbose=`n`, max level/times is 3)")
 	f.BoolVar(&globalOptions.NoLock, "no-lock", false, "do not lock the repository, this allows some operations on read-only repositories")
 	f.BoolVarP(&globalOptions.JSON, "json", "", false, "set output mode to JSON for commands that support it")
-	f.StringVar(&globalOptions.CacheDir, "cache-dir", "", "set the cache `directory`. (default: use system default cache directory)")
+	f.StringVar(&globalOptions.CacheDir, "cache-dir", os.Getenv("RESTIC_CACHE_DIR"), "set the cache `directory`. (default: use system default cache directory)")
```

Comment:
Although this seems to change the behavior of all commands, it actually has only an effect on the `check` command. All other commands already use the `DefaultDir()` function to determine the cache directory location. And that function also processes the environment variable.

So it's probably better to implement the change here instead:
https://github.com/restic/restic/blob/553ea36ca62c870d71cd31d4091c86d0ec984e82/cmd/restic/cmd_check.go#L148

I'd also prefer to not hardcode the environment variable name in different places in restic, but rather keep it limited to `internal/cache/dir.go`. That is please add a helper function there.